content/build/cache/invalidation.md

```diff
@@ -0,0 +1,79 @@
+---
+title: Build cache invalidation
+description: Dig into the details abouw how cache invalidation works for Docker's build cache
+keywords: build, buildx, buildkit, cache, invalidation, cache miss
+---
+
+When building an image, Docker steps through the instructions in your
+Dockerfile, executing each in the order specified. For each instruction, Docker
+checks whether it can reuse the instruction from the build cache.
+
+## General rules
+
+The basic rules of build cache invalidation are as follows:
+
+- Starting with a parent image that's already in the cache, the next
+  instruction is compared against all child images derived from that base
+  image to see if one of them was built using the exact same instruction. If
+  not, the cache is invalidated.
+
+- In most cases, simply comparing the instruction in the Dockerfile with one
+  of the child images is sufficient. However, certain instructions require more
+  examination and explanation.
+
+- For the `ADD` and `COPY` instructions, the modification time and size file
+  metadata is used to determine whether cache is valid. During cache lookup,
+  cache is invalidated if the file metadata has changed for any of the files
+  involved.
+
+- Aside from the `ADD` and `COPY` commands, cache checking doesn't look at the
+  files in the container to determine a cache match. For example, when processing
+  a `RUN apt-get -y update` command the files updated in the container
+  aren't examined to determine if a cache hit exists. In that case just
+  the command string itself is used to find a match.
+
+Once the cache is invalidated, all subsequent Dockerfile commands generate new
+images and the cache isn't used.
+
+If your build contains several layers and you want to ensure the build cache is
+reusable, order the instructions from less frequently changed to more
+frequently changed where possible.
+
+## RUN instructions
+
+The cache for `RUN` instructions isn't invalidated automatically between builds.
+Suppose you have a step in your Dockerfile to upgrade all the software packages
+in your Debian-based image to the latest version:
+
+```dockerfile
+RUN apt-get update && apt-get upgrade -y
```

Comment:
I don't think this is recommended to upgrade all packages (cc @tianon). Can you put an example with a simple package like `curl`?

Reply:
Maybe also switch to Alpine based image that has packages updated more often than Debian based distro.

Reply:
Switched to `apk` and `curl` for this one